app/models/services/locators/doi_paper_locator.rb

```diff
@@ -15,10 +16,22 @@ def find_or_import_paper
     if paper_attributes
       response = crossref.resource.response
       paper = Paper.create paper_attributes
+
+      return nil if paper.errors.any?
+
       paper.api_import_responses.create(json: JSON.parse(response.body), source_uri: response.uri.to_s)
       return paper
     else
       return nil
     end
   end
+
+  def valid?
+    # Stolen from http://blog.crossref.org/2015/08/doi-regular-expressions.html
+    is_doi = locator_id.match(/^10.\d{4,9}\/[-._;()\/:A-Z0-9]+$/i)
+
+    errors << "\"#{locator_id}\" does not match DOI format. Ex: \"10.1371/journal.pone.0001897\"" unless is_doi
```

Comment:
this is great, I don't know why, but I really like getting this error message when I write "poop" in the DOI add field.  The site just feels more legit now.